python_modules/dagster/dagster/core/definitions/decorators/solid_decorator.py

```diff
@@ -373,6 +374,20 @@ def resolve_checked_solid_fn_inputs(
                 used_inputs.add(param.name)
 
     undeclared_inputs = explicit_names - used_inputs
+
+    if decorator_name == "@graph":
+        undeclared_nothing_inputs = [
+            inp.name
+            for inp in explicit_input_defs
+            if inp.dagster_type.kind == DagsterTypeKind.NOTHING and inp.name in undeclared_inputs
```

Comment:
can do dagster_type.is_nothing

Reply:
though to sandys point - the actual type being set to Nothing is increasingly rare and if we change the APIs may never proc. 

maybe a better approach would be to have to split the error message we throw below for undeclared inputs into different messages for composition functions (graph/job) and compute function (op, resource, ). 

Taking a closer look at the message its kind of incorrect for graph/job anyway as it mentions the context parameter.